drivers/hmis/app/controllers/hmis/base_controller.rb

```diff
@@ -54,4 +55,25 @@ def set_app_user_header
   def set_git_revision_header
     response.headers['X-git-revision'] = Git.revision
   end
+
+  def impersonating?
+    true_hmis_user != current_hmis_user
+  end
+
+  # for mixins
+  def current_app_user
+    current_hmis_user
+  end
+
+  def authenticate_user!
+    raise 'authenticate_user called in HMIS controller. Did you mean authenticate_user?'
+  end
+
+  def current_user
+    raise 'current_user called in HMIS controller. Did you mean current_hmis_user?'
```

Comment:
There are tons of calls to `current_user` in the mutation code - won't those break? It looks like technically they are referring to a `current_user` thats defined by the `GraphqlApplicationHelper` and comes from the graphql context, so maybe that's why its fine? 🧐

Reply:
I think that's fine, we map the user in the GraphqlController:
```ruby
context = {  current_user: current_hmis_user }
```